islands/opModal.tsx

```diff
@@ -36,10 +38,41 @@ export default function OpModal(
     grpcToken: string;
   },
 ) {
+  const displayType = opModal.value?.displayType;
+  const setDisplayName = () => {
+    if (displayType === "operation") {
+      return opModal.value?.audience?.operationName;
+    } else if (displayType === "service") {
+      return opModal.value?.audience?.serviceName;
+    } else if (displayType === "component") {
+      return opModal.value?.audience.componentName;
+    }
+  };
+
+  const iconDisplay = () => {
```

Comment:
If this display function logic gets re-used across the nodes and drawer you could make them react components.